system_test/m1_facilitator/Deposit.ts

```diff
@@ -0,0 +1,246 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Account } from 'web3-eth-accounts';
+import Mosaic from 'Mosaic';
+import Web3 from 'web3';
+
+import AddressHandler from '../common/AddressHandler';
+import Faucet from '../common/Faucet';
+import Utils from '../common/Utils';
+import Logger from '../../src/common/Logger';
+
+interface Balance {
+  [key: string]: number;
+}
+
+export default class Deposit {
+  public static async depositSystemTest(): Promise<void> {
+    const config = await Utils.getConfig();
+    const {
+      depositorCount,
+      concurrencyCount,
+      iterations,
+      pollingInterval,
+      timeoutInterval,
+    } = config.testData.deposit;
+    const originWsEndpoint = config.chains.origin.wsEndpoint;
+    const auxiliaryWsEndpoint = config.chains.auxiliary.wsEndpoint;
+    const originChainId = config.chains.origin.chainId;
+
+    const messageHashes: string[] = [];
+
+    const initialOriginAccountBalance: Balance = {};
+    const expectedOriginAccountBalance: Balance = {};
+    const initialAuxiliaryAccountBalance: Balance = {};
+    const finalAuxiliaryAccountBalance: Balance = {};
+
+    let testDepositorAccounts = [];
+    let totalDepositorAccounts: any[] = [];
+
+    for (let i = 0; i < iterations; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      testDepositorAccounts = await AddressHandler.getRandomAddresses(
+        depositorCount,
+        concurrencyCount,
+      );
+
+      // eslint-disable-next-line no-await-in-loop
+      await Faucet.fundAccounts(testDepositorAccounts, originChainId);
```

Comment:
If you are using the same set of keys every time, there is no need to fund each time and return remaining to the faucet. 

Fund when necessary.